src/librustc_traits/lowering/mod.rs

```diff
@@ -157,13 +158,27 @@ crate fn program_clauses_for<'a, 'tcx>(
     tcx: TyCtxt<'a, 'tcx, 'tcx>,
     def_id: DefId,
 ) -> Clauses<'tcx> {
+    // FIXME(eddyb) this should only be using `def_kind`.
```

Comment:
Do you mean introducing `DefKind::Impl` or something else?

Reply:
Yeah, I think `DefKind` should be able to describe everything with a `DefId`.